Add: MXP self-healing recovery mechanism

MAVRICK-1 opened the pull request:
# MXP Self-Healing Recovery Mechanism

## Summary

This PR implements an automatic detection and recovery system for malformed MXP (Mud eXtension Protocol) parsing issues that cause screen output to freeze, addressing [GitHub issue #7896](https://github.com/Mudlet/Mudlet/issues/7896).

## Problem

MUDs often send improperly formatted MXP tags with unescaped characters (`<`, `>`, `&`) that cause Mudlet's MXP parser to get stuck in infinite loops, freezing screen output. This particularly affects new players connecting to MUDs with problematic MXP implementations.

## Solution

###  **Error Detection**
- Tracks consecutive failed MXP parsing attempts
- Triggers recovery mechanism after 1000 failed attempts
- Captures problematic content for user display

###  **Self-Healing Dialog**
- User-friendly dialog explaining the issue
- Shows sample of problematic content
- Three recovery options:
  - **Disable MXP** (recommended) - Turns off MXP processing
  - **Continue with MXP** - Resets error state, keeps MXP enabled
  - **Not Now** - Dismisses dialog without changes

###  **Automatic Recovery**
- Seamless fallback to text processing when MXP is disabled
- Prevents screen output freezing
- Maintains profile settings for future sessions



## Technical Details

### Error Detection Logic
```cpp
if (!mMxpTagBuilder.accept(ch) && mMxpTagBuilder.isInsideTag() && !mMxpTagBuilder.hasTag()) {
    mTagStuckCounter++;
    mProblematicContent += ch;
    
    if (mTagStuckCounter > 1000) {
        // Trigger recovery mechanism
    }
}
```

### Recovery Integration
The recovery system integrates into the main MXP processing loop in `TBuffer.cpp`, checking for stuck parsing and presenting the recovery dialog when issues are detected.

## Testing

Tested with various malformed MXP scenarios:
- Unescaped `<` and `>` characters in ASCII art
- Incomplete MXP tags (e.g., `<font color=red` without closing `>`)
- Mixed problematic content with valid MXP
- Recovery dialog functionality and user option handling

## Benefits

1. **Prevents Screen Freezing** - No more stuck MXP parsing blocking game output
2. **User-Friendly** - Clear explanation and recovery options
3. **Backward Compatible** - Existing MXP functionality unchanged
4. **Minimal Disruption** - Users can choose their preferred recovery method
5. **Educational** - Explains MXP issues to help user understanding

## Screenshots


https://github.com/user-attachments/assets/d4a51836-0606-48b0-a228-a3a2f85a0969


## Impact

This change particularly benefits new players who encounter MUDs with problematic MXP implementations, preventing the frustrating experience of frozen screen output that could drive them away from the game.

## Related Issues

Fixes #7896 - Detect MXP issues and offer self-healing options

/claim #7896

## Checklist

- [x] Code compiles without warnings
- [x] Functionality tested with various malformed MXP scenarios
- [x] Dialog interface tested and user-friendly
- [x] Backward compatibility maintained
- [x] No breaking changes to existing MXP functionality
- [x] Build system updated for new files

MAVRICK-1 committed:
Add MXP self-healing recovery mechanism

Implements automatic detection and recovery from malformed MXP parsing
issues that cause screen output to freeze, addressing GitHub issue #7896.

Features:
- Error detection when MXP parser gets stuck (>1000 failed attempts)
- User-friendly recovery dialog with clear explanation
- Three recovery options: Disable MXP, Continue, or Dismiss
- Automatic fallback to text processing when MXP is disabled
- Prevents screen output freezing from malformed MXP tags

Technical Implementation:
- Enhanced TMxpProcessor with error state tracking
- New dlgMxpRecovery dialog for user interaction
- Integration in TBuffer.cpp for seamless error handling
- Maintains backward compatibility with existing MXP functionality

Tested with various malformed MXP scenarios including unescaped
characters, incomplete tags, and problematic ASCII art content.

MAVRICK-1 committed:
Fix: Add missing newline at end of dlgMxpRecovery source files

MAVRICK-1 commented:
cc @vadi2 @mpconley

MAVRICK-1 commented:
cc @vadi2 @mpconley

mpconley commented:
The approach looks reasonable. Since I could not find snapshots, I installed the branch locally and was looking for places to test it. You need to add in a mudlet.pro changes like those in CMakeLists.txt (easy enough).

MAVRICK-1 committed:
Add MXP recovery dialog files to mudlet.pro

As requested by maintainer, added dlgMxpRecovery.cpp and dlgMxpRecovery.h to the SOURCES and HEADERS sections in mudlet.pro to match the changes in CMakeLists.txt

MAVRICK-1 commented:
@mpconley now it's okay ?

MAVRICK-1 committed:
Merge branch 'development' into feature/mxp-self-healing-recovery

vadi2 committed:
Merge branch 'development' into feature/mxp-self-healing-recovery

vadi2 reviewed (approved):
Looks perfect to me, will let others review.

Kebap commented:
Test result by @[taranion](https://github.com/taranion) in #8072: 

> Tested with Mudlet 4.19.1-testing-pr8042-b3622240, but the problem remains. Only disabling MXP and reconnecting helps - or closing the profile.

mpconley commented:
Please check out this feedback from the [Mudlet Discord](https://discord.com/channels/283581582550237184/283582068334526464/1407647864859004939) where they tested out the PR but did not have luck with an `&`.

MAVRICK-1 commented:
@mpconley 
```
if (!mMxpTagBuilder.accept(ch) && mMxpTagBuilder.isInsideTag() && !mMxpTagBuilder.hasTag()) {
    mTagStuckCounter++;
    mProblematicContent += ch;
    
    if (mTagStuckCounter > 1000) {
        // Trigger recovery mechanism
    }
}
```

try reducing to  50

MAVRICK-1 commented on src/TMxpProcessor.cpp:
```diff
@@ -153,8 +153,20 @@ void TMxpProcessor::disable()
+        mTagStuckCounter++;
+        mProblematicContent += ch;
+        
+        if (mTagStuckCounter > 1000) {
```
i have to just reduce this

MAVRICK-1 commented:
@mpconley @vadi2 @Kebap i have tested with @taranion server its working fine for me

## this pr fix 

https://github.com/user-attachments/assets/dab52cc7-a91b-4838-846e-9154acf99b40


## current main

<img width="647" height="804" alt="grafik" src="https://github.com/user-attachments/assets/4e60f7e9-c9cb-4192-8452-7c6c7e2ab2f9" />

there may be some issue with snapshot 
https://make.mudlet.org/snapshots/9446e7/Mudlet-4.19.1-testing-pr8042-b3622240-linux-x64.AppImage.tar 

the changes are not reflecting here

vadi2 commented:
The snapshot is for commit b3622240 which is the latest commit available

MAVRICK-1 commented:
@vadi2, then what could be the issue that's causing it not to work for @taranion but to work on mine? i am also on the latest commit

MAVRICK-1 commented:
@vadi2 the snapshot is working for me.

vadi2 commented:
I get the same using the snapshot, which is the broken behaviour using the latest commit:

<img width="1588" height="1600" alt="image" src="https://github.com/user-attachments/assets/9b185f82-1330-4f8a-ab94-66e95165b2a2" />

To test it I logged on and did `skills`.

MAVRICK-1 commented:
Strange 🤔

MAVRICK-1 commented:
https://github.com/user-attachments/assets/9d42cabf-a417-490f-8ac4-dbc10dac5233

@vadi2  latest snapshot

vadi2 commented:
This was tackled elsewhere now, but thank you for taking the time to create the contribution.
